ignition-server/main.go

```diff
@@ -134,52 +172,17 @@ func run(ctx context.Context, opts Options) error {
 			http.Error(w, "Unauthorized", http.StatusUnauthorized)
 			return
 		}
-		if !bytes.Equal(decodedToken, token) {
-			http.Error(w, "Unauthorized", http.StatusUnauthorized)
-			return
-		}
 
-		// Build proxy request.
-		proxyReq, err := http.NewRequest("GET", targetURL, nil)
-		if err != nil {
-			// Send 500 response.
-			log.Printf("Server internal error: %v", err)
-			http.Error(w, fmt.Sprintf("Server internal error: %v", err), http.StatusInternalServerError)
-		}
-
-		// Copy original Headers into proxy request.
-		for k, vv := range r.Header {
-			for _, v := range vv {
-				proxyReq.Header.Add(k, v)
-			}
-		}
-
-		// Send proxy request.
-		client := &http.Client{
-			Timeout: 5 * time.Second,
-		}
-		resp, err := client.Do(proxyReq)
-		if err != nil {
-			// Send 503 response.
-			log.Printf("Service unavailable: %v", err)
-			http.Error(w, fmt.Sprintf("Service unavailable: %v", err), http.StatusServiceUnavailable)
+		payload, ok := controllers.PayloadStore.Get(string(decodedToken))
+		if !ok {
+			log.Printf("Payload not found")
+			http.NotFound(w, r)
```

Comment:
I suggest we use `http.Error(w, "Unauthorized", http.StatusUnauthorized)` here to avoid giving the caller too much information about the nature of the failure (e.g. malicious users)